### ISSUE TITLE
Thank you for your work on Seniman, I am interested, How I can help?

### ISSUE THREAD

**Issue by rubiagatra** [senimanjs/seniman#3]
Hello @garindra, I am interested in contributing to Seniman JS. Can I help with something?

For testing or CI / CD?

Waiting for your response

Thankyou

**Comment by garindra** [comment]
Hi Doni -- thanks for the interest!

What do you think about adding TypeScript `d.ts` type files?

**Comment by rubiagatra** [comment]
Let me tinkering it first, I'll get back to you later

**Comment by garindra** [comment]
Goal is to allow TypeScript users to have its features when using Seniman -- I assume with the `d.ts` type files this is achieved. 

Internally within Seniman I'd like to stick with pure JS for now.

**Comment by rubiagatra** [comment]
Yes, I agree. I have experience contributing to fastify with JS code and still maintaining the types.

I will create the draft PR soon

**Comment by ncrypthic** [comment]
Working with seniman is really fun, it simplify a lot of things. I've tried to use seniman with an existing typescript react project with custom `.d.ts` files to satisfy the typescript compiler. And it doubled the fun! (For me at least 😁).

The type definition is mostly reverse engineered from the implementation here. Though I just make the definition only for types and functions that I use in the project.

I think adding a proper type definition to be used by typescript users will be a big win!

I would also contribute to add the types.

**Comment by garindra** [comment]
@rubiagatra, @ncrypthic might give you a good starting point if you haven't gone far into it
